modules/internal/ChapelRange.chpl

```diff
@@ -1778,25 +1778,45 @@ private proc isBCPindex(type t) param do
       if _low > _high then return false;
     }
 
-    if ! this.isAligned() || ! other.isAligned()
+    if ! this.isAligned()
       then return false;
 
     return true;
   }
 
   inline proc range.chpl_boundsCheck(other: range(?e,?b,?s))
   {
-    if ! this.isAligned() || ! other.isAligned()
+    if ! this.isAligned()
       then return false;
 
-    var boundedOther = new range(
+    var boundedOther = if other.isAligned() then
+                        new range(
                           idxType, boundKind.both, other.strides,
                           if other.hasLowBound() then other._low else _low,
                           if other.hasHighBound() then other._high else _high,
-                          other._stride, other._alignment);
+                          other._stride, other._alignment)
+                      else if this.hasPosNegUnitStride() then
+                        new range(
+                          idxType, boundKind.both, other.strides,
+                          if other.hasLowBound() then other._low else _low,
+                          if other.hasHighBound() then other._high else _high,
+                          other._stride, 0)
```

Comment:
Let us cast the above zero to this.strType, just to be on the safe side.

Reply:
That seems unnecessarily paranoid to me (since param downcasting will let 0 become any integral type and I think we only support integral stride types?).  Are you anticipating a problem I'm not seeing?

Reply:
> That seems unnecessarily paranoid to me (since param downcasting will let 0 become any integral type and I think we only support integral stride types?). Are you anticipating a problem I'm not seeing?

It is perhaps easier to make this change than to explain it. Still:

The range initializer invoked here does the following, see https://github.com/chapel-lang/chapel/blob/2a6141eb6aa17c7222e4a7a528c178c509381930/modules/internal/ChapelRange.chpl#L311 

where `alignmentValue` is the formal that the zero is passed to. The formal is unconstrained generic so its type becomes `int(64)`. This formal is not a param. Further, if `this` of the chpl_boundsChecking method has idxType=int(8), then in the initializer `this._alignment` will also have the type `int(8)`. The assignment into int(8) from int(64) will generate an unnecessary compilation error.